pilot/pkg/xds/cds.go

```diff
@@ -15,53 +15,50 @@
 package xds
 
 import (
-	"time"
-
-	cluster "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
-	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
-
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pilot/pkg/networking/util"
-	v3 "istio.io/istio/pilot/pkg/xds/v3"
+	"istio.io/istio/pkg/config/schema/gvk"
+	"istio.io/istio/pkg/config/schema/resource"
 )
 
-// clusters aggregate a DiscoveryResponse for pushing.
-func cdsDiscoveryResponse(response []*cluster.Cluster, noncePrefix string) *discovery.DiscoveryResponse {
-	out := &discovery.DiscoveryResponse{
-		// All resources for CDS ought to be of the type Cluster
-		TypeUrl: v3.ClusterType,
-
-		// Pilot does not really care for versioning. It always supplies what's currently
-		// available to it, irrespective of whether Envoy chooses to accept or reject CDS
-		// responses. Pilot believes in eventual consistency and that at some point, Envoy
-		// will begin seeing results it deems to be good.
-		VersionInfo: versionInfo(),
-		Nonce:       nonce(noncePrefix),
-	}
-
-	for _, c := range response {
-		out.Resources = append(out.Resources, util.MessageToAny(c))
-	}
-
-	return out
+type CdsGenerator struct {
+	Server *DiscoveryServer
 }
 
-func (s *DiscoveryServer) pushCds(con *Connection, push *model.PushContext, version string) error {
-	pushStart := time.Now()
-	defer func() { cdsPushTime.Record(time.Since(pushStart).Seconds()) }()
+var _ model.XdsResourceGenerator = &CdsGenerator{}
 
-	rawClusters := s.ConfigGenerator.BuildClusters(con.proxy, push)
+// Map of all configs that do not impact CDS
+var skippedCdsConfigs = map[resource.GroupVersionKind]struct{}{
+	// TODO: investigate if this is correct when PILOT_FILTER_GATEWAY_CLUSTER_CONFIG is set
```

Comment:
it is not. that flag reads vs to determine clusters

Reply:
Right so if a new virtual service is added doesn't that mean we need to update cds potentially?

Reply:
Ok will handle in a followup - don't want to change behavior too much here as its already a large PR